src/backend/tests/reimbursement-requests.test.ts

```diff
@@ -625,4 +626,28 @@ describe('Reimbursement Requests', () => {
       expect(newReimbursement).toStrictEqual(reimbursementTransformer(reimbursementMock));
     });
   });
+
+  describe('Edit Vendor Tests', () => {
+    test('Throws error if user isnt an admin', async () => {
+      await expect(
+        ReimbursementRequestService.editVendors('I Love Benny', GiveMeMyMoney.vendorId, wonderwoman)
+      ).rejects.toThrow(new AccessDeniedAdminOnlyException('only Admins can edit vendors'));
+    });
+
+    test('Vendor Name already exists', async () => {
+      vi.spyOn(prisma.vendor, 'findUnique').mockResolvedValue(PopEyes);
+      await expect(ReimbursementRequestService.editVendors('CHICKEN', GiveMeMyMoney.vendorId, batman)).rejects.toThrow(
+        new HttpException(400, 'vendor name already exists')
+      );
+    });
+
+    test('Successfuly changes Vendors name', async () => {
+      vi.spyOn(prisma.vendor, 'update').mockResolvedValue(KFC);
+      vi.spyOn(prisma.vendor, 'findUnique').mockResolvedValue(null);
+
+      const vendor = await ReimbursementRequestService.editVendors('kfc', PopEyes.vendorId, batman);
+
+      expect(vendor.name).toBe('kfc');
```

Comment:
Can we also add an expected line where it checks the Prisma.vendor.update has been called 1 time? This is a way we test if our endpoint works as we expected by using Prisma ORM.

For example, we can refer to `toggles successfully` test case in projects.tests.ts